src/services/user-payload.ts

```diff
@@ -29,6 +31,21 @@ export class UserPayload {
     this.username =  `${this.username}${randomNumber.toString()}`;
   }
 
+  /**
+   * Retrieves the user type from the place. 
+   * If roles are present, this method returns an empty string.
+   * If roles are not present, it returns the user type defined.
+   * 
+   * When the type is sent alongside roles to CHT, roles are ignored.
+   * In such cases, this method returns an empty string to prioritize roles.
+   *
+   * @param place - The place object containing creation details and type information.
+   * @returns A string representing the user type.
+   */
+  public getUserType(place: Place): string {
```

Comment:
i find this flow a bit confusing... it's kinda awkward that roles is sometimes in control and sometimes it is type. 

could we just always set `roles` and never set `type`?